tools/frrcommon.sh.in

```diff
@@ -335,7 +335,7 @@ if [ -z "$FRR_PATHSPACE" ]; then
 	load_old_config "/etc/sysconfig/frr"
 fi
 
-if { declare -p watchfrr_options 2>/dev/null || true; } | grep -q '^declare \-a'; then
+if { declare -p watchfrr_options 2>/dev/null || true; } | grep -q '^declare -a'; then
```

Comment:
I believe `grep -q -- '^declare -a'` should work too?

Reply:
yes, but since the `'^declare -a'` is a constant string not starting with `-` and not a variable (`grep -q -- "$declare_array_pattern"`), it's IMO not needed to protect it / force an end of option-scanning.